src/contrib/cluster/Akka.Cluster.Tools/PublishSubscribe/Internal/Topics.cs

```diff
@@ -116,22 +151,23 @@ protected bool DefaultReceive(object message)
                 default:
                     foreach (var subscriber in Subscribers)
                         subscriber.Forward(message);
+
+                    // no subscribers
+                    if (Subscribers.Count == 0 && SendToDeadLettersWhenNoSubscribers)
```

Comment:
The bugfix: @eaba pointed out that this actor never was programmed to publish to deadletters when the subscriber count is zero, so we do this after the `foreach` loop (arbitrary) if we're configured for it. This causes the reproduction spec to now pass.